spec/shared/json_convertible_spec.rb

```diff
@@ -102,5 +119,35 @@ module FastlaneCI
       expect(object.mock_json_attribute.one_attribute).to eql("Hello")
       expect(object.mock_json_attribute.other_attribute).to eql(Time.at(0))
     end
+
+    it "Allows to decode from a dictionary object when it has an array typed attribute" do
+      allow(MockArrayJSONConvertible).to receive(:attribute_to_type_map).and_return({ :@mock_json_array_attribute => MockJSONConvertible })
+      dictionary_object = { "mock_json_array_attribute" => [{ "one_attribute" => "Hello", "other_attribute" => Time.at(0) }, { "one_attribute" => "World", "other_attribute" => Time.at(10) }] }
+      object = MockArrayJSONConvertible.from_json!(dictionary_object)
+      expect(object.mock_json_array_attribute.kind_of?(Array)).to eql(true)
+      expect(object.mock_json_array_attribute.length).to eql(2)
+      expect(object.mock_json_array_attribute[0].kind_of?(MockJSONConvertible)).to eql(true)
+      expect(object.mock_json_array_attribute[1].kind_of?(MockJSONConvertible)).to eql(true)
+      expect(object.mock_json_array_attribute[0].one_attribute).to eql("Hello")
+      expect(object.mock_json_array_attribute[1].one_attribute).to eql("World")
+    end
+
+    it "Allows to decode from a dictionary object using custom mapping per object in array property" do
+      allow(MockArrayJSONConvertible).to receive(:map_enumerable_type).and_return(mock_object)
```

Comment:
I think we need a test that ensures `:map_enumerable_type` actually what it is supposed to do (instead of mocking it here)

Reply:
I mean, we should keep this test, but add another that also tests `: map_enumerable_type:`